File: lib/domain/shared/dtos/referencable_list_row.rb
# frozen_string_literal: true

module Domain
  module Shared
    module Dtos
      # 一覧行: ドメイン record のみ（API JSON は entity をそのまま出す）。
      class ReferencableListRow
        attr_reader :record

        def initialize(record:)
          @record = record
        end
      end
    end
  end
end

File: crates/agrr-domain/src/shared/dtos/referencable_list_row.rs
/// Ruby: `Domain::Shared::Dtos::ReferencableListRow`
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct ReferencableListRow<R> {
    pub record: R,
}

impl<R> ReferencableListRow<R> {
    pub fn new(record: R) -> Self {
        Self { record }
    }
}
